### ISSUE TITLE
Fails to parse `foo: no good`

### ISSUE THREAD

**Issue by softmoth** [Leont/yamlish#17]
The document `foo: no good` won't parse. The document `foo: so good` will.
In `token element()`, `<value=inline>` is failing, and it apparently won't backtrack to attempt `<value=plain>`.
I made a stripped-down test case:
```
grammar Grammar {
    rule TOP { <element> }
    token plain { \N* }
    token inline { yes | no }
    token element {
        <value=inline> || <value=plain>
    }
}

use Data::Dump;
for 'no', 'no bar', 'xno bar' {
    note "\n$_";
    Grammar.parse($_);
    dd $/;
    dd $/<element>.Str;
}
```
It parses "no" and "xno bar" fine, but "no bar" fails. Even if `plain`, `inline` and `element` are all `regex` instead of `token`, it still fails. Only if both `TOP` and `element` are changed to `regex` will it parse.

**Comment by Leont** [comment]
> It parses "no" and "xno bar" fine, but "no bar" fails. Even if plain, inline and element are all regex instead of token, it still fails. Only if both TOP and element are changed to regex will it parse.

Another recent bug made me suspect that I'd need to turn more rules from token to regex, this confirms that suspicion. I guess I'll have to turn a lot of tokens into regexes…

**Comment by softmoth** [comment]
I wish I understood it better. It seems to me that just making `regex element` would be correct, but obviously I'm not understanding how it works. Making everything `regex` seems a pity. Good luck!

**Comment by Leont** [comment]
> I wish I understood it better. It seems to me that just making regex element would be correct, but obviously I'm not understanding how it works.

@Zoffixznet explained it to me on IRC, I don't think I need to change everything, just things that aren't anchored basically.

**Comment by Leont** [comment]
This has been fixed in 0.0.5
